.github/workflows/ci.yml

```diff
@@ -39,35 +53,48 @@ jobs:
         run: go vet github.com/couchbase/sync_gateway/...
 
   test:
+    needs: run-bootstrap
     env:
       GO111MODULE: off
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
```

Comment:
I think we want to disable fail-fast.

Means that if one item in the matrix fails we still run the other. ie. if one OS fails we still want to test the other. When I say 'fail' this will only occur if the whole thing errors, ie. a panic as opposed to a test failure but still might be nice

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast